tensorflow_datasets/testing/dataset_builder_testing.py

```diff
@@ -261,6 +271,19 @@ def test_download_and_prepare_as_dataset(self):
     else:
       self._download_and_prepare_as_dataset(self.builder)
 
+    if not self.SKIP_CHECKSUMS:
+      with self._subTest("url_checksums"):
+        self._test_checksums()
+
+  def _test_checksums(self):
+    urls = []
+    filepath = self._get_checksums_filepath()
+    with tf.io.gfile.GFile(filepath, "rb") as f:
+      for line in f.readlines():
+        urls.append(line.split()[0].decode("utf-8"))
+
+    self.assertEqual(set(self.URLS), set(urls))
```

Comment:
The user is likely to be confused about this error. Could you raise a more explicit error message ?